### PR TITLE
Feature/test stall

### PR DESCRIPTION
**Pull request by xross**
Added tests for STALLing or HALTing an Endpoint.

~Note, test_stall_epready is a directed test for a current issue and so currently fails.~

Now passing.

I also added dummy threads running in one of the tests, it eventually will need adding to all tests.

Also some tidies.

### CODE REVIEW

File: tests/test_stall_epready.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        UsbTransaction(
+            session,
+            deviceAddress=address,
+            endpointNumber=ep,
+            endpointType="BULK",
+            direction="OUT",
+            dataLength=pktLength,
+            halted=True,
+        )
+    )
```
Comment (DaveAtkinsonXMOS) [inline tests/test_stall_epready.py:50]: As above, we're sending the same packet twice. Just checking this isn't a cut 'n' paste error ;-)

File: tests/usb_packet.py [hunk 2]
```diff
@@ -59,28 +59,31 @@
...
 
 from usb_event import UsbEvent
+import usb_phy
+
 
 USB_DATA_VALID_COUNT = {"FS": 39, "HS": 0}
 
 # In USB clocks
-RX_TX_DELAY = 20
+# RX_TX_DELAY = 20
```
Comment (DaveAtkinsonXMOS) [inline tests/usb_packet.py:68]: Is this done with?

File: tests/usb_packet.py [hunk 3]
```diff
@@ -59,28 +59,31 @@
...
 RXA_END_DELAY = 2  # Pad delay not currently simulated in xsim for USB or OTP, so add this delay here
 RXA_START_DELAY = 5  # Taken from RTL sim
-RX_RX_DELAY = 40
+# RX_RX_DELAY = 40
```
Comment (DaveAtkinsonXMOS) [inline tests/usb_packet.py:71]: And is this done with also?

**Review by DaveAtkinsonXMOS — COMMENTED** [review]
Spotted two bits of commented-out code, and just noted a couple of things that were duplicates which are most probably correct, I just thought I'd check...

File: tests/test_stall_basic.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        UsbTransaction(
+            session,
+            deviceAddress=address,
+            endpointNumber=ep,
+            endpointType="BULK",
+            direction="OUT",
+            dataLength=pktLength,
+            halted=True,
+        )
+    )
```
[the remark this answers is not in the document]
Comment (xross) [reply tests/test_stall_basic.py:48]: Just sending the same packet twice to check the DUT doesn't work just the once :)

File: tests/usb_packet.py (hunk 2 above)
Comment (xross) [reply tests/usb_packet.py:68]: Fixed

File: tests/usb_packet.py (hunk 3 above)
Comment (xross) [reply tests/usb_packet.py:71]: Fixed

**Review by DaveAtkinsonXMOS — APPROVED** [review]
Good to go...

**Pull request merged**